### PR TITLE
add hacbs release-service apiexport

### PR DESCRIPTION
**Pull request by johnbieren**
Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

**Comment by johnbieren** [comment]
Depends on https://github.com/redhat-appstudio/release-service/pull/116

**Comment by Michkov** [comment]
> Depends on [redhat-appstudio/release-service#116](https://github.com/redhat-appstudio/release-service/pull/116)

I don't think so, [redhat-appstudio/release-service#116](https://github.com/redhat-appstudio/release-service/pull/116) depends on this PR

File: components/hacbs/release-service/overlays/kcp-stable/apiexport.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+apiVersion: apis.kcp.dev/v1alpha1
+kind: APIExport
+metadata:
+  name: release-service
```
Comment (Michkov) [inline components/hacbs/release-service/overlays/kcp-stable/apiexport.yaml:5]:
```
$ kustomize build components/hacbs/release-service/overlays/kcp-stable/
Error: no matches for Id apis.kcp.dev_v1alpha1_APIExport|release-service|release-service; failed to find unique target for patch apis.kcp.dev_v1alpha1_APIExport|release-service
```

APIExport is named 'release' in the release-service project

Comment (johnbieren) [reply components/hacbs/release-service/overlays/kcp-stable/apiexport.yaml:5]: another good catch. changing it in the release-service project to be `release-service`

Comment (johnbieren) [reply components/hacbs/release-service/overlays/kcp-stable/apiexport.yaml:5]: changed in https://github.com/redhat-appstudio/release-service/pull/116

**Review by Michkov — APPROVED** [review]

**Comment by hongweiliu17** [comment]
@johnbieren @Michkov Do you know why the release-service component is added under `components/hacbs`, not `components`?  Because I notice it is mentioned the component is added like `components/(component-name)` in step 1 and 2 of https://github.com/redhat-appstudio/infra-deployments/#how-to-add-your-own-component, will be our components considered as `hacba/*` or other reason? Thank you!

**Comment by Michkov** [comment]
> @johnbieren @Michkov Do you know why the release-service component is added under `components/hacbs`, not `components`? Because I notice it is mentioned the component is added like `components/(component-name)` in step 1 and 2 of https://github.com/redhat-appstudio/infra-deployments/#how-to-add-your-own-component, will be our components considered as `hacbs/*` or other reason? Thank you!

@hongweiliu17 currently it's helper for APIExports templating, since some APIExports can have dependency on other APIExports which are not created yet, then they are pre-created just to generate IdentityHashes to be used by the APIExport. And for that I need to know that the service is HACBS so the APIExport is pre-created inside redhat-hacbs workspace.
Related magic code -> https://github.com/redhat-appstudio/infra-deployments/blob/4523b5821bbab53ec1c26bf8dc7181413cd86f20/hack/preview.sh#L96-L131

**Comment by johnbieren** [comment]
@Michkov can we merge this?

**Pull request merged**